routers/api/packages/cargo/cargo.go

```diff
@@ -48,7 +48,7 @@ func apiError(ctx *context.Context, status int, obj any) {
 
 // https://rust-lang.github.io/rfcs/2789-sparse-index.html
 func RepositoryConfig(ctx *context.Context) {
-	ctx.JSON(http.StatusOK, cargo_service.BuildConfig(ctx.Package.Owner))
+	ctx.JSON(http.StatusOK, cargo_service.BuildConfig(ctx.Package.Owner, ctx.Package.Descriptor.Repository.IsPrivate))
```

Comment:
That's not correct. For the sparse index the visibility of the linked package-repo is not relevant. `Descriptor` should be `nil` for this endpoint too.

I think the correct value would be something like:
`setting.Service.RequireSignInView || ctx.Package.Owner.Visibility != structs.VisibleTypePublic`
Instance requires login or owner is not public

Reply:
> `Descriptor` should be `nil` for this endpoint too.

The failed tests confirm that.

Reply:
Applied the fix